### PR TITLE
Improve the "Debug Console" to add functionality and fix bugs

### LINKED ISSUE #943: Improve the "Debug Console" page
While using the "Debug Console" of the Arduino Agent, a few improvement opportunities were found:
- Save the selection for the "show list commands", so that it will persist page refresh
- Improve the "Enable list command" checkbox to allow 3 different settings:
  - `Hide "list" commands`: like having the checkbox unchecked
  - `Show "list" commands inline`: like having the checkbox checked
  - `Show "list" commands separately`: will show a new section where you see the result of the last "list" command, independently from the other logs
- Fix a regression where the "list" lines where hidden, but the actual content (the part with the board data) was always shown, no matter the checkbox state.

### PR DESCRIPTION
**Pull request by Xayton**
Changes:
- Improved the CSS code and JS code
- Added the support for the "show 'list'" options as a select
- Add an additional section to show only the "list" commands, when the user selects such option
- Fixed the code detecting "list" messages, as it was missing cases where the JSON content had a "Ports" property but its value was null instead of an empty array.

Closes #943

### CODE REVIEW

**Review by alessio-perugini — APPROVED** [review]
Super :rocket: 

I see that we're using a very old version of jquery, what do you think if we remove it and use just vanilla js?

File: home.html [hunk 1]
```diff
@@ -1,38 +1,53 @@
...
+            var input = $('#input');
+            var log = document.getElementById('log');
+            var logList = document.getElementById('log-list');
+
+            var autoscroll = document.getElementById('autoscroll');
+            var messages = [];
+            const MESSAGES_MAX_COUNT = 2000;
+
+            // Check if the list visibility setting is saved in the localStorage.
+            const LOCAL_STORAGE_KEY = "ArduinoAgentListVisibility";
```
Comment (umbynos) [inline home.html:23]: :open_mouth: noice!

**Review by umbynos — APPROVED** [review]
Nice Work!

**Comment by Xayton** [comment]
@alessio-perugini @umbynos jQuery has been removed, and code simplified a bit. Can be easily tested by building the agent locally.

**Review by alessio-perugini — APPROVED** [review]
Sweet :star_struck: .
Tested on ubuntu working :rocket:

**Pull request merged**